docker/STAR/1.0.0-alpha/sort_star_input.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+
+
+def sort_fastqs(read_one_fastqs, read_two_fastqs):
+    read1_names = [fastq.split('/')[-1] for fastq in read_one_fastqs]
+    read2_names = [fastq.split('/')[-1] for fastq in read_two_fastqs]
+    
+    for i in range(1, len(read1_names)):
```

Comment:
What I'm doing is reordering the full paths, `read_<one/two>_fastqs`, based on the alphanumeric sorting of the basenames, `read<1/2>_names`.

Using `sorted()` on the paths will be wrong, because it will sort based on the cromwell directories, not the read name. And using `sorted()` on the basenames won't allow me to map those index changes back to the file paths.

`enumerate()` won't work because I'm iterating over `len() - 1`. I don't want to iterate over every element.

This whole thing absolutely needs better documentation. I wasn't even sure if it would solve the problem with the MLL samples when I wrote it, so didn't bother documenting till I knew that what I thought the problem was, actually was the problem. Might have thrown out this whole script.

Reply:
What about something like this? I would this this is more canonical/simple.

```python
filepaths = [
    "/Users/cmcleod/baz",
    "/Users/afrantz/foo",
    "/Users/jsunny/bar"
]

basenames = [
    "baz",
    "foo",
    "bar"
]

sorted_filepaths, sorted_basenames = zip(*sorted(zip(filepaths, basenames), key = lambda item: item[1]))
print(f"Sorted filepaths: {sorted_filepaths}")
print(f"Sorted basenames: {sorted_basenames}")
```

Reply:
IMO that's much less clear. Someone just casually looking at the code would need a much better understanding of Python to follow that one liner as opposed to the insertion sort function I wrote. And this whole script takes less than a second to run, so IMO any performance differences are negligible.

However that is leveraging Python builtins, which is more professional. It admittedly felt silly rewriting insertion sort 😂.

I think since performance is moot it comes down to which is more "presentable" code. I can see it going either way, readability vs professionalism; not exactly sure who the audience for this code is.

Reply:
After taking a look at that code in context I like it more than my implementation. Going to commit your suggested change here.

Reply:
Sounds good! In this case, I'd argue the shorter code actually makes it more understandable. This might be a prime place for a comment for those not familiar argument unpacking.